intake_geopandas/geopandas.py

```diff
@@ -70,28 +82,68 @@ def __init__(self, urlpath, bbox=None,
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
+
+        # warn if using fsspec caching and same_names not True for zip files
+        if 'cache' in self.urlpath and 'zip' in self.urlpath:
+            same_names = False  # default
+            # find different same_names setting
+            for c in ['blockcache', 'filecache', 'simplecache']:
+                if c in self.storage_options:
+                    if 'same_names' in self.storage_options[c]:
+                        same_names = self.storage_options[c]['same_names']
+            if not same_names:
+                warnings.warn(
+                    'Need same_names = True for local caching of `zip` files.'
+                )
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
         Open dataset using geopandas and use pattern fields to set new columns.
         """
+        if 'cache' in self.urlpath:  # new caching
+            url = fsspec.open_local(self.urlpath, **self.storage_options)
+            if not os.path.exists(url):
+                # explicitly loading the file if not cached
+                url = fsspec.open(self.urlpath, **self.storage_options)
+        elif self.cache:  # old caching
+            self.cache[0].load(self.urlpath)
+            if isinstance(self.cache[0], str):
+                url = self.cache[0]
+            else:
+                url = self.urlpath
+        else:
+            url = self.urlpath
+        # opening locally cached files, geopandas expects local zip paths with `zip://`
+        # see https://geopandas.org/io.html
+        if os.path.exists(url):
```

Comment:
actually I want to check whether url is note remote. is there a nice fsspec way to check this?

Reply:
Not really; there isn't a clear meaning to "remote" anyway, aside from "not local file". The latter will have a protocol of "file:", or none.
You could do `fsspec.open` and look for FileNotFoundError to check existence in general, but it might only throw when you actually read from the file.

Reply:
implemented this now with `requests`